### PR TITLE
Fix i2c busy / bus busy handling

### LINKED ISSUE #34: I2C not working anymore
After last changes in I2C, it seems the logic of the busy_wait macro is not correct.
I've been able to patch it locally and will push a clean proposal tomorrow (@thejpster asked me to file an issue).

Basically, checking for BusBusy everywhere is not correct as it will always be set between START and STOP, so during a transaction, we should not return an error (that's what I'm currently seeing).

Previous code was basically adapted from stm32f30x-hal and I see that the busy_wait macro has been changed (no more flag as parameter) and is not really busy_waiting on anything but the BUSY bit. The macro is now wrapped in `loop{}`. Was is not better to have the busy wait inside the macro ? Maybe there's something I'm not seeing here.

The original code was doing things like:
- wait for `busbsy` to be clear
- do something
- wait for `busy` to be clear
- do something
....

I see that previous code was wrongly looking at all `mcs` bits without waiting for `busy` to be clear, but maybe the cleaning went too far ? At least with my patch that reverts back the logic above, my application is working correctly again.

### PR DESCRIPTION
**Pull request by dkm**
Restore previous logic where a transaction starts by waiting for the controler
to be ready and the bus available and then only wait for the controller to be ready
at intermediate step (not the bus to be available as it will be busy).
Modify the i2c_busy_wait to implement this behavior.

fix #34

### CODE REVIEW

**Comment by thejpster** [comment]
To be clear, this basically reverts https://github.com/rust-embedded-community/tm4c-hal/pull/29/commits/4d50753d21331a667a97bc585f83fe6b5d7cf26c

**Comment by dkm** [comment]
Not true ! :) It keeps most of these changes, only it won't raise an error when `busbsy` is set everytime the macro is used, only when the extra argument is provided (eg. at beg of transaction).
It still spins on `busy` to be clear before reading other bits (which the patch you link fixes).

**Comment by thejpster** [comment]
I appreciate the clarification. I find Github diff mode quite hard to grok :/

**Comment by dkm** [comment]
No problem ! And I may still have overlooked something, so feel free to question my choices :) I can still say that previous try to use i2c always failed with the `BusBusy` error since the @dtwood's changes and now it works correctly for hours. Not a clear proof it fixes something, but at least, it works here :)

File: tm4c-hal/src/i2c.rs [hunk 1]
```diff
@@ -60,26 +62,45 @@ macro_rules! i2c_busy_wait {
             .write(|w| unsafe { w.cntl().bits((1_000 >> 4) as u8) });
 
         let mcs = loop {
-            let mcs = $i2c.mcs.read();
+                let mcs = $i2c.mcs.read();
```
Comment (thejpster) [inline tm4c-hal/src/i2c.rs:65]: I think this extra indent is unnecessary?

File: tm4c-hal/src/i2c.rs [hunk 2]
```diff
@@ -60,26 +62,45 @@ macro_rules! i2c_busy_wait {
...
-        } else if mcs.arblst().bit_is_set() {
-            Err(Error::Arbitration)
-        } else if mcs.datack().bit_is_set() {
-            Err(Error::DataAck)
-        } else if mcs.adrack().bit_is_set() {
-            Err(Error::AdrAck)
-        } else {
-            Ok(())
-        }
+                return Err(Error::Timeout)
```
Comment (thejpster) [inline tm4c-hal/src/i2c.rs:74]: Also extra indentation here. It's annoying rustfmt won't format inside a macro.

Comment (dkm) [reply tm4c-hal/src/i2c.rs:74]: Sorry, didn't know rustfmt does not indent macro, I was convinced that everything was correct after using it.

**Comment by dkm** [comment]
Great, thanks ! I can say that before your changes, i2c was really unstable, after your fix it does not work at all, and the combination of your fix and these changes makes it work flawlessly on my app. It is simple, so can't say it's a very robust test, but it is testing basic things...

**Pull request merged**